### PR TITLE
Gekidou Sentry Installation

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary

This is the first PR related to the Sentry feature.

#### ACHIEVED
In here, the following have been tested: 

1. successful sourcemap upload for Android
2. successful sourcemap upload for iOS
3. the error event shows the line number for Android
4. the error event shows the line number for iOS
5. integration with the react-native-navigation lib for Performance monitoring

#### PENDING
1. Test on iOS CircleCi Pipeline - Probably the provisioning profile expired and thus failing the build
2. Update the `sentry context` on CircleCI to reflect the values from Sentry MM iOS+Android projects
3. Update  the .`circleci/config.yml` file to include the `sentry context` along with the updated properties.

For iOS extensions, Sentry do catch the errors as describe on [this issue](https://github.com/getsentry/sentry-cocoa/issues/1656#issuecomment-1263336691)

I also created a patch for the 'sentry.gradle' file for it not to include invalid file paths for the sourcemap upload.  That's the reason why it was failing and I also [created a PR](https://github.com/getsentry/sentry-react-native/pull/2589) towards their repository.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -84,12 +84,12 @@ project.ext.react = [
 apply from: "../../node_modules/react-native/react.gradle"
 apply from: "../../node_modules/react-native-vector-icons/fonts.gradle"
 
-if (System.getenv("SENTRY_ENABLED") == "true") {
-    project.ext.sentryCli = [
+project.ext.sentryCli = [
```
Comment (avinashlng1080) [inline android/app/build.gradle:87]: For some reason, if this block is inside the 'if' block at line 92, the 'upload sourcemap' task wouldn't run.  Moving it outside and having the environment variable 'SENTRY_ENABLD' set in the bash_profile worked for the sourcemap upload

File: app/utils/error_handling.ts [hunk 2]
```diff
@@ -23,8 +24,9 @@ import {logWarning} from './log';
 class JavascriptAndNativeErrorHandler {
     initializeErrorHandling = () => {
         initializeSentry();
-        setJSExceptionHandler(this.errorHandler, false);
 
+        // NOTE: The below methods have been disabled as they interferred with Sentry error reporting.
```
Comment (avinashlng1080) [inline app/utils/error_handling.ts:28]: The below methods have been disabled as they interferred with Sentry error reporting.

File: android/app/build.gradle [hunk 3]
```diff
@@ -84,12 +84,15 @@ project.ext.react = [
 apply from: "../../node_modules/react-native/react.gradle"
 apply from: "../../node_modules/react-native-vector-icons/fonts.gradle"
 
-if (System.getenv("SENTRY_ENABLED") == "true") {
-    project.ext.sentryCli = [
+
+/** Disable the below if you are not using Sentry */
+def SENTRY_ENABLED = true
```
Comment (enahum) [inline android/app/build.gradle:89]: This should be set to an environment variable

File: android/app/build.gradle (hunk 1 above)
Comment (enahum) [reply android/app/build.gradle:90]: This was working just fine, why is this not working?

File: android/app/build.gradle [hunk 4]
```diff
@@ -144,7 +147,7 @@ android {
         applicationId "com.mattermost.rnbeta"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 425
+        versionCode 426
```
Comment (enahum) [inline android/app/build.gradle:150]: This should not be paart of this PR

File: app/utils/error_handling.ts (hunk 2 above)
Comment (enahum) [reply app/utils/error_handling.ts:28]: We used the js handler to forward the exception to sentry, why don't we need that anymore?

The native handler uses a third parameter that if you set as true, then the sentry error handler should work just fine.

Perhaps read the README of the library and inspect its source code to understand more

File: assets/base/config.json [hunk 5]
```diff
@@ -13,9 +13,9 @@
     "MobileNoticeURL": "https://about.mattermost.com/mobile-notice-txt/",
     "RudderApiKey": "",
 
-    "SentryEnabled": false,
-    "SentryDsnIos": "",
-    "SentryDsnAndroid": "",
+    "SentryEnabled": true,
+    "SentryDsnIos": "https://8166acfeeced43de98bc11f217d871c5@o1347733.ingest.sentry.io/6734065",
+    "SentryDsnAndroid": "https://8166acfeeced43de98bc11f217d871c5@o1347733.ingest.sentry.io/6734065",
```
Comment (enahum) [inline assets/base/config.json:18]: This should not be modified, see the dev docs on how to override this values in your local config

File: fastlane/Gemfile.lock [hunk 6]
```diff
@@ -66,7 +66,7 @@ GEM
     faraday_middleware (1.2.0)
       faraday (~> 1.0)
     fastimage (2.2.6)
-    fastlane (2.209.1)
+    fastlane (2.210.0)
```
Comment (enahum) [inline fastlane/Gemfile.lock:69]: Why the upgrade of fastlane in this pr?

File: ios/Mattermost.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -891,6 +892,20 @@
 			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n# This output is used by Xcode 'outputs' to avoid re-running this script phase.\necho \"SUCCESS\" > \"${SCRIPT_OUTPUT_FILE_0}\"\n";
 			showEnvVarsInLog = 0;
 		};
+		64399EEDB3C64470AC8AD28F /* Upload Debug Symbols to Sentry */ = {
```
Comment (enahum) [inline ios/Mattermost.xcodeproj/project.pbxproj:895]: Wasn't this phase already defined?

File: ios/Mattermost/Info.plist [hunk 8]
```diff
@@ -37,7 +37,7 @@
 		</dict>
 	</array>
 	<key>CFBundleVersion</key>
-	<string>425</string>
+	<string>426</string>
 	<key>ITSAppUsesNonExemptEncryption</key>
```
Comment (enahum) [inline ios/Mattermost/Info.plist:41]: Not needed in this PR

File: ios/bundleReactNative.sh [hunk 9]
```diff
@@ -6,13 +6,17 @@ export NODE_OPTIONS=--max_old_space_size=12000
 export BUNDLE_COMMAND="bundle"
 export ENTRY_FILE="index.ts"
 
+# Disable the below if you are not using Sentry
+export SENTRY_ENABLED=true
```
Comment (enahum) [inline ios/bundleReactNative.sh:10]: This should be an env var, not to be set here

File: ios/makeSentryProperties.sh [hunk 10]
```diff
@@ -1,5 +1,8 @@
 #!/bin/sh
 
+# Disable the below if you are not using Sentry
+export SENTRY_ENABLED=true
```
Comment (enahum) [inline ios/makeSentryProperties.sh:4]: Same comment about the env var

File: ios/uploadDebugSymbols.sh [hunk 11]
```diff
@@ -1,12 +1,16 @@
 #!/bin/sh
 
+# Disable the below if you are not using Sentry
+export SENTRY_ENABLED=true
```
Comment (enahum) [inline ios/uploadDebugSymbols.sh:4]: Here too

**Review by enahum — CHANGES_REQUESTED** [review]

File: ios/Mattermost/Info.plist (hunk 8 above)
Comment (avinashlng1080) [reply ios/Mattermost/Info.plist]: Corrected

File: ios/bundleReactNative.sh (hunk 9 above)
Comment (avinashlng1080) [reply ios/bundleReactNative.sh]: Removed

File: ios/makeSentryProperties.sh (hunk 10 above)
Comment (avinashlng1080) [reply ios/makeSentryProperties.sh]: Removed.

File: ios/uploadDebugSymbols.sh (hunk 11 above)
Comment (avinashlng1080) [reply ios/uploadDebugSymbols.sh]: Removed.

File: android/app/build.gradle (hunk 4 above)
Comment (avinashlng1080) [reply android/app/build.gradle]: Corrected

File: assets/base/config.json (hunk 5 above)
Comment (avinashlng1080) [reply assets/base/config.json]: By the dev docs, is it [this one](https://docs.mattermost.com/configure/configuration-settings.html) ? For the mobile app, did we take a subset of the server config and added it in our code base?

File: ios/Mattermost.xcodeproj/project.pbxproj (hunk 7 above)
Comment (avinashlng1080) [reply ios/Mattermost.xcodeproj/project.pbxproj:895]: Just double checked it and it seems that it got duplicated.  Removing it now.

File: package.json [hunk 12]
```diff
@@ -30,7 +30,7 @@
     "@react-navigation/bottom-tabs": "6.3.2",
     "@react-navigation/native": "6.0.11",
     "@rudderstack/rudder-sdk-react-native": "1.4.1",
-    "@sentry/react-native": "4.2.2",
+    "@sentry/react-native": "4.3.0",
```
Comment (enahum) [inline package.json:33]: There is a newer version

File: ios/uploadDebugSymbols.sh [hunk 13]
```diff
@@ -6,7 +6,8 @@ if [[ "${SENTRY_ENABLED}" = "true" ]]; then
 	./makeSentryProperties.sh
 
 	export SENTRY_PROPERTIES=sentry.properties
-	../node_modules/@sentry/cli/bin/sentry-cli upload-dsym
+../node_modules/@sentry/cli/bin/sentry-cli upload-dif "$DWARF_DSYM_FOLDER_PATH"
```
Comment (enahum) [inline ios/uploadDebugSymbols.sh:9]: Why this change?

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/utils/error_handling.ts (hunk 2 above)
Comment (enahum) [inline app/utils/error_handling.ts:28]: Why do they interfere, this was working together just fine in v1

File: app/utils/sentry.ts [hunk 14]
```diff
@@ -33,7 +34,20 @@ export function initializeSentry() {
         return;
     }
 
-    Sentry.init({dsn, ...Config.SentryOptions});
+    Sentry.init({
```
Comment (enahum) [inline app/utils/sentry.ts:37]: this is fine, but I do not see any changes in the user & extra context being made, is that going to be addressed?

File: android/app/build.gradle (hunk 1 above)
Comment (avinashlng1080) [reply android/app/build.gradle]: reverted

File: app/utils/error_handling.ts (hunk 2 above)
Comment (avinashlng1080) [reply app/utils/error_handling.ts:28]: reverted

File: android/app/build.gradle (hunk 3 above)
Comment (avinashlng1080) [reply android/app/build.gradle]: reverted

File: ios/uploadDebugSymbols.sh (hunk 13 above)
Comment (avinashlng1080) [reply ios/uploadDebugSymbols.sh:9]: new cli functions

File: app/utils/sentry.ts (hunk 14 above)
Comment (avinashlng1080) [reply app/utils/sentry.ts:45]: added in app/screens/home/channel_list/channel_list.tsx

**Comment by avinashlng1080** [comment]
Closing in favor of #6726
